Enhancement: Flow run results lists

znicholasbrown opened the pull request:
This PR:

- improves mock flow run and task run results artifact description generation (they now more closely resemble action descriptions)
- breaks flow run and task run results into their own lists
- improves some `ArtifactCard` designs
- adds result descriptions to the default slot of `ArtifactCard`s 
- adds an interval to the artifact subscription so results update in real time
- persists a user's list vs grid preference (just using local storage right now but that seemed light enough)
- adds some new demo sections

![Screen Shot 2023-03-15 at 9 53 59 PM](https://user-images.githubusercontent.com/27291717/225489540-39da151f-48ef-4784-9ec3-1fff7d752cd0.png)

![Screen Shot 2023-03-15 at 9 54 08 PM](https://user-images.githubusercontent.com/27291717/225489538-22770d28-047d-4779-831f-7408a17d585f.png)

pleek91 commented on src/localization/locale/en.ts:
```diff
@@ -88,6 +88,8 @@ export const en = {
   info: {
     artifact: 'Artifact',
     noResults: 'No tracked results',
+    flowRunResults: 'Flow run',
+    taskRunResults: 'Task runs',
```
Nit: As we start adding more of these we need to be careful about naming. If we ever add a second language the names will be important. I'd literally name these `flowRun` and `taskRuns` as these are pretty generic labels and aren't restricted to results.

pleek91 commented:
Couple things I noticed while looking at the deploy preview on my phone. The grid view overlaps and doesn't scroll well. And in the list view the breadcrumbs overflow the card

![image](https://user-images.githubusercontent.com/6200442/225499153-4c07c561-2325-458d-9313-d253785b28b6.png)
![image](https://user-images.githubusercontent.com/6200442/225499093-3a52122d-2af0-4309-a4f6-9c5bde4c82cd.png)

znicholasbrown commented:
@pleek91 which browser are you seeing that in? I've got a pretty nice looking version in chrome desktop at 330px:

![Screenshot 2023-03-16 at 11 51 41 AM](https://user-images.githubusercontent.com/27291717/225676089-0dd66b42-610b-4315-a607-9bff84e88d03.png)

zhen0 commented:
Small note here that the deploy preview looks good on my iphone now.  Looks like you're digging in further though? Let me know if I can help keep this one moving.

znicholasbrown commented:
Thanks @zhen0; the remember your selection feature doesn't seem to be working in the deploy preview but is working in a regular ole build. I've added some fallbacks so that we can feel comfortable releasing just in case that continues to fail downstream. 

@pleek91 I've added a cache-key wrapper
